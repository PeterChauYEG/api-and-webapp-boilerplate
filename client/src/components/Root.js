import React, { Component } from 'react'

// import css
import 'font-awesome/css/font-awesome.css'
import '../CSS/helpers.css'
import '../CSS/Root.css'

// import components
import Minions from './Minions'

class Root extends Component {
  componentWillMount() {
    const {
      actions,
      minions,
      user,
    } = this.props
    
    // check if minions are loaded
    if (minions.minions.length === 0) {

        /* load minions*/
        actions.getMinionsRequest(user.token)        
      
    }

  }

  render() {
    const {
      actions,
      minions,
    } = this.props

    return (
      <div className="Root bg-lightest-blue pa3 helvetica">
        {console.log(this.props)}

      <Minions
        {...this.props}
        actions={actions}
        minions={minions} />

      </div>
    );
  }
}

export default Root