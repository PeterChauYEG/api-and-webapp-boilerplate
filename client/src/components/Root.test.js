import React from 'react'

// import test dependencies 
import renderer from 'react-test-renderer'

// import component
import Root from './Root'

// render test
test('renders without crashing', () => {
  
  // declare prop
  const actions = {
    getMinionsRequest: () => {},
    toggleMinionForm: () => {},
  }
  const minions = {
    minions: [
      
    ],
  }
  const user = {
    token: null,
  }
  
  // render component
  const component = renderer.create(
    <Root
      actions={actions} 
      minions={minions}
      user={user}
      />
  )

  // render DOM tree
  const tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});
