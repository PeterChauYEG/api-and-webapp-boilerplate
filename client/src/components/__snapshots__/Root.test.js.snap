exports[`test renders without crashing 1`] = `
<div
  className="Root bg-lightest-blue pa3">
  <div
    className="Minions pa4">
    <h1>
      Minions
    </h1>
    <div>
      <p>
        There\'s no minions yet. You should add one!
      </p>
    </div>
    <div
      className="HoverButton bg-light-blue"
      onClick={[Function]}>
      <i
        className="fa fa-plus fa-2x" />
    </div>
  </div>
</div>
`;
